pkg/apis/operator/v1alpha1/types_extension.go

```diff
@@ -83,6 +83,10 @@ type ExtensionDeploymentSpec struct {
 	// An empty list means that all seeds are selected.
 	// +optional
 	SeedSelector *metav1.LabelSelector `json:"seedSelector,omitempty"`
+	// InjectGardenKubeconfig controls whether a kubeconfig to the garden cluster should be injected into workload
+	// resources.
+	// +optional
+	InjectGardenKubeconfig *bool `json:"injectGardenKubeconfig,omitempty"`
```

Comment:
Any plans to do the same with the admission deployment where the Garden Kubeconfig is automatically injected as well?

Reply:
I think we can assume that admission components always need a garden kubeconfig in order to properly validate things, no?

Reply:
Maybe for provider extensions but I wouldn't expect admission controllers of generic extensions to always require a garden client. They often only validate their own provider config against the shoot object.

(Can be addressed in a separate PR)

Reply:
@timuthy thanks for pointing this out!